### PR TITLE
[Edifice] Test de la fonctionnalité d'ajout de fonctions à un utilisateur

### PR DESCRIPTION
**Pull request by juniorode**
# Description

Test de la fonctionnalité d'ajout de fonctions à un utilisateur

## Type of change

Please check options that are relevant.

- [x] Chore (PATCH)
- [ ] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [ ] New feature (MINOR)

## Which packages changed?

Please check the name of the package you changed

- [ ] admin
- [ ] app-registry
- [ ] archive
- [ ] auth
- [ ] cas
- [ ] common
- [ ] communication
- [ ] conversation
- [ ] directory
- [ ] feeder
- [ ] infra
- [ ] portal
- [ ] session
- [ ] test
- [x] tests
- [ ] timeline
- [ ] workspace

# Reminder

- Security flaws
- Performance impacts (think bulk !)
- Unit tests were replayed
- Unit tests were added and/or changed
- I have updated the reminder for the version including my modifications

- [x] All done ! :smiley:

### CODE REVIEW

File: tests/src/test/scenarios/features/functions.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
+@functions-feature
+Feature: Functions feature
+  Make sure that functions are securely managed
+
+  Scenario: ADML users cannot give themselves ADML rights on a structure then do not administrate
```
Comment (mariusode) [inline tests/src/test/scenarios/features/functions.feature:5]: typo : the**y** do not administrate

File: tests/src/test/scenarios/features/functions.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Given A structure "s1"
+      And A structure "s2"
+      And A user "user1" on the structure "s1"
+      And User "user1" is ADML on the structure "s1"
+    When "user1" adds the ADML function to "user1" on structure "s2"
+    Then I get a 401 response
+
+  Scenario: ADML users can give ADML rights to other user on the structure they administrates
```
Comment (mariusode) [inline tests/src/test/scenarios/features/functions.feature:13]: typo : 
- other user**s**
- they administrate_

File: tests/src/test/scenarios/features/functions.feature [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    Then I get a 200 response
+
+  Scenario: Non-ADML users cannot give ADML rights to other users on a structure
+    Given A structure "s1"
+    And A user "user1" on the structure "s1"
+    And A user "user2" on the structure "s1"
+    When "user1" adds the ADML function to "user2" on structure "s1"
+    Then I get a 401 response
+
+  Scenario: ADMC users can give ADML rights to other user on any structure
```
Comment (mariusode) [inline tests/src/test/scenarios/features/functions.feature:28]: other user**s**

**Review by mariusode — COMMENTED** [review]
just little typos

**Review by mariusode — APPROVED** [review]

File: tests/src/test/scenarios/features/functions.feature (hunk 1 above)
Comment (juniorode) [reply tests/src/test/scenarios/features/functions.feature]: OK

File: tests/src/test/scenarios/features/functions.feature (hunk 3 above)
Comment (juniorode) [reply tests/src/test/scenarios/features/functions.feature]: OK

**Pull request merged**